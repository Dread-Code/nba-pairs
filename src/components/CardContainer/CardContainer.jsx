import React from 'react'
import { Grid } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import CardPairs from '../CardPairs/CardPairs'
import AlertCard from '../AlertCard/AlertCard'

const CardContainer = ({ players }) => (
  <div>
    {players?.length > 0 ? (
      <Grid doubling columns={3}>
        {players.map((playersPair, key) => (
          <Grid.Column width="2" key={key + 1}>
            <CardPairs playersPair={playersPair} />
          </Grid.Column>
        ))}
      </Grid>
    ) : (
      <AlertCard error="No matches found" />
    )}
  </div>
)

CardContainer.propTypes = {
  players: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)).isRequired
}

export default CardContainer
