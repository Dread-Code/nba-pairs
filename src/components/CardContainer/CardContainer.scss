.cardpaircontainer {
  padding: 0.5vw;
  background: #f2711c;
  border-radius: 2vw;
  margin: 0.5vw;
}
