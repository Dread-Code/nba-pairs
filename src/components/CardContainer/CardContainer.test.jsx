import React from 'react'
import { shallow } from 'enzyme'
import CardContainer from './CardContainer'

const playersMock = [
  [
    {
      first_name: 'Nate',
      h_in: '69',
      h_meters: '1.75',
      last_name: 'Robinson'
    },
    {
      first_name: 'Chucky',
      h_in: '71',
      h_meters: '1.8',
      last_name: 'Atkins'
    }
  ],
  [
    {
      first_name: 'Mike',
      h_in: '70',
      h_meters: '1.78',
      last_name: 'Wilks'
    },
    {
      first_name: 'Brevin',
      h_in: '70',
      h_meters: '1.78',
      last_name: 'Knight'
    }
  ]
]

describe('CardContainer', () => {
  test('1) should match snapshoot', () => {
    const wrapper = shallow(<CardContainer players={playersMock} />)
    expect(wrapper).toMatchSnapshot()
  })
  test('2) should return 2 elements ', () => {
    const wrapper = shallow(<CardContainer players={playersMock} />)
    expect(wrapper.find('GridColumn').getElements().length).toBe(playersMock.length)
  })
  test('3) should return card alert if recive a empty array', () => {
    const wrapper = shallow(<CardContainer players={[]} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GridColumn').getElements().length).toBe(0)
    expect(wrapper.find('AlertCard').getElements().length).toBe(1)
  })
})
