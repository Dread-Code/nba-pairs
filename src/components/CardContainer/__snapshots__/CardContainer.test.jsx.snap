// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer 1) should match snapshoot 1`] = `
<div>
  <Grid
    columns={3}
    doubling={true}
  >
    <GridColumn
      key="1"
      width="2"
    >
      <CardPairs
        playersPair={
          Array [
            Object {
              "first_name": "Nate",
              "h_in": "69",
              "h_meters": "1.75",
              "last_name": "Robinson",
            },
            Object {
              "first_name": "Chucky",
              "h_in": "71",
              "h_meters": "1.8",
              "last_name": "Atkins",
            },
          ]
        }
      />
    </GridColumn>
    <GridColumn
      key="2"
      width="2"
    >
      <CardPairs
        playersPair={
          Array [
            Object {
              "first_name": "Mike",
              "h_in": "70",
              "h_meters": "1.78",
              "last_name": "Wilks",
            },
            Object {
              "first_name": "Brevin",
              "h_in": "70",
              "h_meters": "1.78",
              "last_name": "Knight",
            },
          ]
        }
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`CardContainer 3) should return card alert if recive a empty array 1`] = `
<div>
  <AlertCard />
</div>
`;
